### PR TITLE
Only check online runner when detecting matching runners in workflows

### PR DESCRIPTION
**Pull request by yp05327**
Mentioned: [#28277](https://github.com/go-gitea/gitea/issues/28277#issuecomment-1831325276)

We should only check online runner when detecting matching runners in workflows, 
as if runner is not online, the workflow will not run.

image

### CODE REVIEW

File: routers/web/repo/actions/actions.go [hunk 1]
```diff
@@ -85,7 +85,9 @@ func List(ctx *context.Context) {
 		}
 		allRunnerLabels := make(container.Set[string])
 		for _, r := range runners {
-			allRunnerLabels.AddMultiple(r.AgentLabels...)
+			if r.IsOnline() {
```
Comment (lunny) [inline routers/web/repo/actions/actions.go]: I think we should change `FindRunnerOptions` to add a new option

Comment (yp05327) [reply routers/web/repo/actions/actions.go]: Done.

File: models/actions/runner.go [hunk 2]
```diff
@@ -153,6 +159,7 @@ type FindRunnerOptions struct {
 	OwnerID       int64
 	Sort          string
 	Filter        string
+	IsOnline      bool
```
Comment (lunny) [inline models/actions/runner.go]: Maybe `util.OptionalBool` is better. Otherwise you cannot select all the runners.

Comment (yp05327) [reply models/actions/runner.go]: Done.

**Review by lunny — APPROVED** [review]

**Review by a1012112796 — APPROVED** [review]

**Pull request merged**